### PR TITLE
Matt Gill -- Reed -- SPRINT -- React Wars

### PR DESCRIPTION
**Pull request by mxxt1**
Doing this preemptively

### CODE REVIEW

File: starwars/src/App.js [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+  //added to CharContainer.js
+
+
+
+
   return (
     <div className="App">
       <h1 className="Header">React Wars</h1>
-    </div>
+      <CharContainer />
```
Comment (MrT3313) [inline starwars/src/App.js:23]: Try to do your axios as close to the top of the app as possible and then pass that data into <CharContainer> as the bridge between the app and the individual cards.

File: starwars/src/components/CharContainer.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+
+const CharContainer = () => {
+
+    const [starChars, setStarChars] = useState([]);
+
+
+    useEffect(() => {
+        axios.get('https://swapi.co/api/people/')
```
Comment (MrT3313) [inline starwars/src/components/CharContainer.js:28]: do this up in the APP and pass into your bridge between app and individual cards

File: starwars/src/components/CharContainer.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    },[])
+
+    // console.log(starChars);
+
+    return(
+        <Container>
+        {starChars.map(character => {
+            return(
+                <CharCard 
+                key={character.name}
```
Comment (MrT3313) [inline starwars/src/components/CharContainer.js:46]: typically the key is an integer so you would need to pass the key / index in the map arguments and then key would be key={key} or key={index}

**Review by MrT3313 — COMMENTED** [review]
STYLING ON FLEEEEEEEEEEK

- try to add some error handling for your n/a values
- good use of passing props

File: starwars/src/components/CharContainer.js (hunk 2 above)
Comment (mxxt1) [reply starwars/src/components/CharContainer.js:28]: D'oh! I initially had state/side-effect/get request in App.js, and moved it to CharContainer.js... and now I can't even remember why!

File: starwars/src/components/CharContainer.js (hunk 3 above)
Comment (mxxt1) [reply starwars/src/components/CharContainer.js:46]: I'll do this next time. I wasn't sure how to handle key in this instance and kept getting "React requires a key..." error when I tried to set it to index, so I went with name and never came back around to troubleshoot it.

**Pull request merged**